common/src/main/java/com/wynntils/services/destination/DestinationService.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.destination;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Service;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DestinationService extends Service {
+    // Map<location, abbreviation>
+    private Map<String, String> destinations;
+
+    public DestinationService() {
+        super(List.of());
+
+        loadDestinations();
+    }
+
+    @Override
+    public void reloadData() {
+        loadDestinations();
+    }
+
+    /**
+     * Get the abbreviation of a location
+     * @param location Full location name of the destination (eg. "Ragni")
+     * @return Abbreviation of the location (eg. "Ra"). Reads from destination json first, if not found returns first + next non-conflicting letter of the location name.
+     */
+    public String getAbbreviation(String location) {
+        if (!destinations.containsKey(location)) {
+            String abbreviation = location.substring(0, 2);
+            int i = 2;
+            while (destinations.containsValue(abbreviation)) {
+                if (location.charAt(i) == ' ') i++;
+                abbreviation = location.charAt(0) + "" + location.charAt(i);
+                i++;
+            }
+            WynntilsMod.warn("DestinationService: No destination found for " + location
+                    + ", using fallback abbreviation " + abbreviation);
+            return abbreviation;
+        }
+        return destinations.get(location);
+    }
+
+    private void loadDestinations() {
+        destinations = new HashMap<>();
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_DESTINATIONS);
+        dl.handleReader(reader -> {
+            Destination result = WynntilsMod.GSON.fromJson(reader, Destination.class);
+            for (Destination.DestinationDetail destination : result.destinations) {
+                destinations.put(destination.location, destination.abbreviation);
```

Comment:
What we generally do when loading anything from net (which is basically async, since the user can trigger a reload) is to use a temp map to load all destinations into, then set the actual field instance. This way, you still have the old destinations if the download fails, plus, you are safe for concurrency.